From 9c4a11d000bee84ca534bbb0734fd244617901c2 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Sat, 7 Feb 2026 21:57:43 -0500
Subject: [PATCH 47/58] x86/cpu: disable bus lock mitigation by default

Set sysctl_sld_mitigate = 0 to avoid performance penalty from
bus lock detection split-lock handling.
---
 arch/x86/kernel/cpu/bus_lock.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/x86/kernel/cpu/bus_lock.c b/arch/x86/kernel/cpu/bus_lock.c
index 981f8b1f0792..0703463bfb0f 100644
--- a/arch/x86/kernel/cpu/bus_lock.c
+++ b/arch/x86/kernel/cpu/bus_lock.c
@@ -44,11 +44,15 @@ static const struct {
 	{ "ratelimit:", sld_ratelimit },
 };
 
 static struct ratelimit_state bld_ratelimit;
 
+#ifdef CONFIG_BUNKERNEL
+static unsigned int sysctl_sld_mitigate = 0;
+#else
 static unsigned int sysctl_sld_mitigate = 1;
+#endif /* CONFIG_BUNKERNEL */
 static DEFINE_SEMAPHORE(buslock_sem, 1);
 
 #ifdef CONFIG_PROC_SYSCTL
 static const struct ctl_table sld_sysctls[] = {
 	{
-- 
2.52.0

